src/components/button/button.js

```diff
@@ -54,34 +54,76 @@ function MdButtonDirective($mdInkRipple, $mdTheming, $mdAria) {
     link: postLink
   };
 
-  function isAnchor(attr) {
-    return angular.isDefined(attr.href) || angular.isDefined(attr.ngHref);
-  }
-  
-  function getTemplate(element, attr) {
-    return isAnchor(attr) ?
-           '<a class="md-button" ng-transclude></a>' :
-           '<button class="md-button" ng-transclude></button>';
-  }
-
   function postLink(scope, element, attr) {
     var node = element[0];
+    var elementHasText = node.textContent.trim();
+
     $mdTheming(element);
     $mdInkRipple.attachButtonBehavior(scope, element);
 
-    var elementHasText = node.textContent.trim();
     if (!elementHasText) {
       $mdAria.expect(element, 'aria-label');
     }
 
-    // For anchor elements, we have to set tabindex manually when the 
-    // element is disabled
-    if (isAnchor(attr) && angular.isDefined(attr.ngDisabled) ) {
-      scope.$watch(attr.ngDisabled, function(isDisabled) {
-        element.attr('tabindex', isDisabled ? -1 : 0);
-      });
+    updateTabIndex(scope, element, attr);
+
+  }
+
+  // ************************************************
+  // Internal Methods
+  // ************************************************
+
+  /**
+   * Publish either an <a/> or <button/> template
+   * @returns {string}
+   */
+  function getTemplate(element, attr) {
+    return isAnchor(attr) ?
+           '<a class="md-button" ng-transclude></a>' :
+           '<button class="md-button" ng-transclude></button>';
+  }
+
+  /**
+   * Should we use an <a/> template
+   * @returns {boolean|*}
+   */
+  function isAnchor(attr) {
+    return angular.isDefined(attr.href) || angular.isDefined(attr.ngHref);
+  }
+
+  /**
+   * Is this a simulate button ?
+   * @returns {boolean}
+   */
+  function isButton(attr) {
+    return !isAnchor(attr) && (attr.role == "button");
+  }
+
+  /**
+   * Update the accessibility `tabindex` property
+   */
+  function updateTabIndex(scope, element, attr) {
+    if ( isAnchor(attr) ) {
+
+      // For anchor elements, we have to set tabindex manually
```

Comment:
Shouldn't `<button>` get the same treatment (change tabindex to -1 when disabled)? To reset to the default, we can either remove the attribute (since they are interactive by default), or reset to the original tabIndex provided by the developer.